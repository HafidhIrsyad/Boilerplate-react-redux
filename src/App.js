import React from 'react';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import Navbar from './components/Navbar';
import AddProduct from './forms/AddProduct';
import AddSeller from './forms/AddSeller';
import HomePage from './pages/index'
import SearchProduct from './pages/SearchProduct';


function App() {
  return (
    <Router>
      <Navbar />
      <Switch>
        <Route path={["/", '/home']} exact key="HomePage" component={HomePage} />
        <Route path="/addProduct" exact key="AddForm" component={AddProduct} />
        <Route path="/addSeller" exact key="AddSeller" component={AddSeller} />
        <Route path="/search" exact key="AddSeller" component={SearchProduct} />
      </Switch>
    </Router>
  );
}

export default App;
