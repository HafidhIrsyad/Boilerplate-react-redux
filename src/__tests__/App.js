import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import App from '../App';

describe('src/App', () => {
  test('render', () => {
    const shallow = new ShallowRenderer();
    const store = {
      dispatch: jest.fn(),
      getState: jest.fn(),
      subscribe: jest.fn(),
    };
    const tree = shallow.render(<App history={{}} store={store} />);
    expect(tree).toMatchSnapshot();
  });
});
