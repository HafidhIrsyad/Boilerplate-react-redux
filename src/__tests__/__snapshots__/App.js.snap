// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/App render 1`] = `
<BrowserRouter>
  <Navbar />
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path={
        Array [
          "/",
          "/home",
        ]
      }
    />
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": Object {
            "destroyOnUnmount": true,
            "enableReinitialize": false,
            "forceUnregisterOnUnmount": false,
            "form": "addProduct",
            "getFormState": [Function],
            "initialValues": Object {
              "deskripsi": "",
              "hargaSatuan": "",
              "nama": "",
              "satuan": "",
              "sellerId": "",
            },
            "keepDirtyOnReinitialize": false,
            "persistentSubmitErrors": false,
            "pure": true,
            "shouldAsyncValidate": [Function],
            "shouldError": [Function],
            "shouldValidate": [Function],
            "shouldWarn": [Function],
            "submitAsSideEffect": false,
            "touchOnBlur": true,
            "touchOnChange": false,
            "updateUnregisteredFields": false,
            "validate": [Function],
          },
          "render": [Function],
        }
      }
      exact={true}
      path="/addProduct"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/addSeller"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/search"
    />
  </Switch>
</BrowserRouter>
`;
